(DOCS-5860)(DOCS-5649) Edit process docs (#15547)
DataDog/integrations-core: process/README.md

## Overview

The Process Check lets you:
- Collect resource usage metrics for specific running processes on any host. For example, CPU, memory, I/O, and number of threads.

> - Use [Process Monitors][1] to configure thresholds for how many instances of a specific process ought to be running and get alerts when the thresholds aren't met (see **Service Checks** below).

Reviewer: ```suggestion
- Use [Process Monitors][1] to configure thresholds for how many instances of a specific process should be running and get alerts when the thresholds aren't met (see **Service Checks** below).
```

## Setup

### Installation

The Process check is included in the [Datadog Agent][2] package, so you don't need to install anything else on your server.

### Configuration

Unlike many checks, the Process check doesn't monitor anything useful by default. You must configure which processes you want to monitor.

While there's no standard default check configuration, here's an example `process.d/conf.yaml` that monitors SSH/SSHD processes. See the [sample process.d/conf.yaml][3] for all available configuration options:

```yaml
init_config:
instances:
- name: ssh
  search_string:
    - ssh
    - sshd
```

**Note**: After you make configuration changes, make sure you [restart the Agent][4].

> Retrieving some process metrics requires either running the Datadog collector as the same user as the monitored process, or using privileged access. For the `open_file_descriptors` metric on Unix platforms, there is an additional configuration option. Setting `try_sudo` to `true` in your `conf.yaml` file allows the Process check to try using `sudo` to collect the `open_file_descriptors` metric. Using this configuration option requires setting the appropriate sudoers rules in `/etc/sudoers`:

Reviewer: ```suggestion
Retrieving some process metrics requires the Datadog collector to either run as the monitored process user or with privileged access. For the `open_file_descriptors` metric on Unix platforms, there is an additional configuration option. Setting `try_sudo` to `true` in your `conf.yaml` file allows the Process check to try using `sudo` to collect the `open_file_descriptors` metric. Using this configuration option requires setting the appropriate sudoers rules in `/etc/sudoers`:
```

```shell
dd-agent ALL=NOPASSWD: /bin/ls /proc/*/fd/
```

### Validation

Run the [Agent's status subcommand][5] and look for `process` under the Checks section.

### Metrics notes

> The following metrics are not available on Linux or OSX:

Reviewer: ```suggestion
The following metrics are not available on Linux or macOS:
```
I feel like this intends to refer to macOS generally, and not the specific version OS X.

> - Process I/O metrics are **not** available on Linux or OSX since the files that the Agent reads (`/proc//io`) are only readable by the process's owner. For more information, [read the Agent FAQ][6].

Reviewer: ```suggestion
- Process I/O metrics are **not** available on Linux or macOS since the files that the Agent reads (`/proc//io`) are only readable by the process's owner. For more information, [read the Agent FAQ][6].
```
